network/network_test.go

```diff
@@ -0,0 +1,150 @@
+package network_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/perlin-network/noise/internal/test/protobuf"
+	"github.com/perlin-network/noise/network"
+)
+
+func TestNodeConnect(t *testing.T) {
+	t.Parallel()
+
+	for _, e := range allEnvs {
+		testNodeConnect(t, e, 10)
+	}
+}
+
+func testNodeConnect(t *testing.T, e env, numNodes int) {
+	te := newTest(t, e)
+	te.startBoostrap(numNodes)
+	defer te.tearDown()
+
+	peers := te.getPeers(te.bootstrapNode)
+
+	count := len(peers)
+	if count != numNodes-1 {
+		t.Errorf("#peers = %d, want %d", count, numNodes-1)
```

Comment:
Can use the assert library to keep the tests more concise:
```
assert.Equalf(t, count, numNodes - 1, "#peers = %d, want %d", count, numNodes-1)
```

Reply:
good point, thanks